devito/passes/clusters/temporal.py

```diff
@@ -0,0 +1,78 @@
+from devito.passes.clusters.utils import cluster_pass
+from devito.ir.support import IterationSpace, SEQUENTIAL
+from devito.ir.support.space import Interval
+from devito.symbolics import xreplace_indices
+from devito.logger import warning
+
+__all__ = ['skewing']
+
+
+@cluster_pass
+def skewing(cluster, *args):
+
+    """
+    Skew the accesses along a SEQUENTIAL Dimension.
+    Example:
+
+    Transform
+
+    for i = 2, n-1
+        for j = 2, m-1
+            a[i,j] = (a[a-1,j] + a[i,j-1] + a[i+1,j] + a[i,j+1]) / 4
+        end for
+    end for
+
+    to
+
+    for i = 2, n-1
+        for j = 2+i, m-1+i
+            a[i,j-i] = (a[a-1,j-i] + a[i,j-1-i] + a[i+1,j-i] + a[i,j+1-i]) / 4
+        end for
+    end for
+    """
+    processed = []
+    # What dimensions do we target?
+    # a) SEQUENTIAL Dimensions
+    skew_dims = []
+    for i in cluster.ispace:
+        if SEQUENTIAL in cluster.properties[i.dim]:
+            skew_dims.append(i.dim)
+
+    skewable = {i.dim for i in cluster.ispace.intervals
+                if not i.dim.symbolic_incr.is_Symbol}
+
+    # Remove candidates
+    mapper, intervals = {}, []
+
+    try:
+        if len(skew_dims) > 1:
+            raise warning("More than 1 dimensions that can be skewed.\
+                          Skewing the first one")
+        skew_dim = skew_dims[0]  # Skew first one
+        intervals.append(Interval(skew_dim, 0, 0))
+        index = intervals.index(Interval(skew_dim, 0, 0))
```

Comment:
this is horrible. You probably wanted:

```
index = Interval(skew_dim, 0, 0)
...
```

but even then, why are you even assigning `index` here? this is getting overridden right after the try-except, not?

Reply:
Needs to be dropped, right.